services/QuillLMS/app/models/checkbox.rb

```diff
@@ -17,4 +17,12 @@ class Checkbox < ActiveRecord::Base
   belongs_to :objective
   belongs_to :user
   validates :objective_id, uniqueness: { scope: :user_id, message: "should only be checked once per user" }
+  after_create :track_onboarding_checklist_analytics
+
+  def track_onboarding_checklist_analytics
+    return if Objective::ONBOARDING_CHECKLIST_NAMES.none?(objective.name)
```

Comment:
Maybe extract the guard check into an instance method in objective.rb:
```
def onboarding_checklist?
  ONBOARDING_CHECKLIST_NAMES.include?(name)
end
```

Then (23) becomes `return unless objective.onboarding_checklist?`